### DISCUSSION TITLE
Restart causes all clients to be logged out

### DISCUSSION THREAD

**Discussion by rikhuijzer** [open-webui/open-webui#1848] [Q&A] [answered]
# Bug Report

First of all, thanks all contributors for making this software. I'm having an instance that I share with some friends and it's been very nice.

## Description

**Bug Summary:**

Every time the open-webui restarts, all logged-in users get logged out. Because the open-webui is often used by multiple devices, this is quite a hassle. I have a script that automatically pulls in the latest open-webui version, but this means that all devices have to log in again every few days. The problem is that when I open open-webui, I'm usually inside some train of thought so having to spend a minute on logging in is very cumbersome. On desktops logging in is quick enough to not be a hassle, but on mobile devices (my primary use-case for open-webui), it is quite cumbersome. 

**Steps to Reproduce:**

Restart the open-webui Docker container.

**Expected Behavior:**

Remain logged in.

**Actual Behavior:**

Have to re-login.

## Environment

- **Open WebUI Version:** 

v0.1.122

- **Ollama (if applicable):** [e.g., 0.1.30, 0.1.32-rc1]

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]

Server runs Ubuntu. Clients run all kinds of OSes including Apple's MacOS and iPadOS.

- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

Firefox on MacOS M2 and Safari on iPhone SE (2022).

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

Docker Compose on Hetzner 2 vCPU with 2 GB RAM.

**Comment by Yanyutin753** [comment]
me too, good idea!

sometimes i set up automatic updates, and then all my friends had to log in again to use it !

**Comment by Yanyutin753** [comment, marked as answer]
I just went back and looked at the code and found that as long as the WEBUI_SECRET_KEY environment variable is defined, I will not have to log in again after restarting.

**Reply by tjbck** [reply]
Indeed, `WEBUI_SECRET_KEY` must be manually set to keep all clients logged in for security reasons.

**Reply by Yanyutin753** [reply]
> Indeed, `WEBUI_SECRET_KEY` must be manually set to keep all clients logged in for security reasons.

thanks，learnded🤓

**Reply by rikhuijzer** [reply]
Thanks! This solved the problem! (I haven't marked the answer as solution because it looks like that option is disabled here in GitHub)

**Comment by Yanyutin753** [comment]
@rikhuijzer You can try ✨
